tequilapi/contract/identity.go

```diff
@@ -129,6 +130,25 @@ type IdentityRegisterRequest struct {
 	ReferralToken *string `json:"token,omitempty"`
 }
 
+// Validate validates fields in request
+func (irr *IdentityRegisterRequest) Validate() *validation.FieldErrorMap {
+	errors := validation.NewErrorMap()
+
+	if okAddress := common.IsHexAddress(irr.Beneficiary); irr.Beneficiary != "" && !okAddress {
+		errors.ForField("beneficiary").Invalid(irr.Beneficiary+" - is not a valid ethereum wallet address")
+	}
+
+	if irr.ReferralToken == nil {
+		if irr.Stake == nil {
+			errors.ForField("stake").Required()
+		} else if irr.Stake.Cmp(big.NewInt(0)) == -1 {
```

Comment:
@chompomonim, @vkuznecovas what is minimal stake for provider to register?

Reply:
There's no artificial limit imposed.